cmd/minishift/cmd/start_preflight_notwin.go

```diff
@@ -0,0 +1,46 @@
+// +build !windows
+
+/*
+Copyright (C) 2017 Red Hat, Inc.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package cmd
+
+import (
+	"fmt"
+	"os/exec"
+	"regexp"
+	"strings"
+)
+
+func checkVBoxInstalled() bool {
+	// check if VBoxManage is really installed by executing a command
+	cmd := exec.Command("VBoxManage", "-v")
+	stdOutStdError, err := cmd.CombinedOutput()
+	if err != nil {
+		return false
+	}
+	stdOut := fmt.Sprintf("%s", stdOutStdError)
+	stdOut = strings.TrimSpace(stdOut)
+	match, err := regexp.MatchString("^[0-9]\\.[0-9]\\.[0-9]+[a-z][0-9]*$", stdOut)
```

Comment:
Do we really have to match the version string. That seems brittle. It should be enough to check the exit code of the command. If it is '0' we should be good. If not something went wrong. Either not installed at all or some other setup issue.

Reply:
Thanks i'll check the exit code than. This crossed my mind too, when they change the versioning format it'll fail.

Reply:
Right.

Reply:
@hferentschik Do you know why the integration tests are failing?

Reply:
I spent some time on this today, couldn't find any easy way to get the exit code.

Reply:
Another thing. The only VirtualBox specific thing in this function is the actual command 'VBoxManage -v'. In its current form, we cannot unit test it at all because we cannot really make any assumptions whether VirtualBox is installed.

If you, however, extract a generic _CommandExecutesSuccessfully(cmd []string) bool_ method you can.

_checkVBoxInstalled()_ becomes:

```Go
func checkVBoxInstalled() bool {
    return CommandExecutesSuccessfully([]string{"VBoxManage", "v"})
}
```

_ CommandExecutesSuccessfully_ could be placed in one of your util packages. Most importantly, it can be tested. We can execute it with:

- CommandExecutesSuccessfully([]string{"thiscommanddoesexist"})
- CommandExecutesSuccessfully([]string{"pwd"})

With a bit of setup code, we can create a temp directory and add a dummy file and make it executable. Then add the temp directory to the PATH and execute _CommandExecutesSuccessfully_ against this path. 

Bottom line, one can test pretty much all possible scenarios of _CommandExecutesSuccessfully_ given confidence in that this method works.

Reply:
The `GetExitCode()` func in `docker/pkg/system` doesn't seem to work as expected, it returns 0 all the time. However i found another piece of code on [stackoverflow](https://stackoverflow.com/a/40770011). Will refactor this and use it, also going to implement  _CommandExecutesSuccessfully()_ but it will also return the exitcode.